src/oss-reproducible/ReproducibleTool.cs

```diff
@@ -0,0 +1,559 @@
+
+            [Option('s', "source-ref", Required = false, Default = "",
+                HelpText = "If a source version cannot be identified, use the specified git reference (tag, commit, etc.).")]
+            public string OverrideSourceReference { get; set; } = "";
+
+            [Option("diff-technique", Required = false, Default = DiffTechnique.Normalized, HelpText = "Configure diff technique.")]
+            public DiffTechnique DiffTechnique { get; set; } = DiffTechnique.Normalized;
+
+            [Option('o', "output-file", Required = false, Default = "", HelpText = "Send the command output to a file instead of standard output")]
+            public string OutputFile { get; set; } = "";
+
+            [Option('d', "show-differences", Required = false, Default = false,
+                HelpText = "Output the differences between the package and the reference content.")]
+            public bool ShowDifferences { get; set; }
+            
+            [Option("show-all-differences", Required = false, Default = false,
+                HelpText = "Show all differences (default: capped at 20), implies --show-differences")]
+            public bool ShowAllDifferences { get; set; }
+
+            [Option('l', "leave-intermediate", Required = false, Default = false,
+                HelpText = "Do not clean up intermediate files (useful for debugging).")]
+            public bool LeaveIntermediateFiles { get; set; }
+        }
+
+        public ReproducibleTool() : base()
+        {
+        }
+
+        /// <summary>
+        /// Main entrypoint for the download program.
+        /// </summary>
+        /// <param name="args">parameters passed in from the user</param>
+        public static async Task Main(string[] args)
+        {
+            ShowToolBanner();
+            Console.WriteLine();
+            var reproducibleTool = new ReproducibleTool();
+            await reproducibleTool.ParseOptions<Options>(args).WithParsedAsync(reproducibleTool.RunAsync);
+        }
+
+        /// <summary>
+        /// Algorithm:
+        /// 0.0 = Worst, 1.0 = Best
+        /// 1.0 => Bit for bit archive match.
+        /// @TODO Refactor this into the individual strategy objects.
+        /// </summary>
+        /// <param name="result"></param>
+        /// <returns></returns>
+        public KeyValuePair<double, string> GetReproducibilityScore(ReproducibleToolResult fullResult)
+        {
+            if (fullResult.Results == null)
+            {
+                return KeyValuePair.Create(0.0, "No reproducibility results were created.");
+            }
+
+            var bestScore = KeyValuePair.Create(0.0, "No strategies were able to successfully derive the package from the source code.");
+
+            foreach (var result in fullResult.Results)
+            {
+                var filesString = result.NumIgnoredFiles == 1 ? "file" : "files";
+
+                if (string.Equals(result.StrategyName, "PackageMatchesSourceStrategy"))
```

Comment:
I think strategyname should be an enum instead of a string.

Reply:
Agree, there's some refactoring that this can be a part of -- there's too much strategy-specific refs from ReproducibleTool - let's include this in that work.